src/_tkagg.cpp

```diff
@@ -101,10 +103,10 @@ static int PyAggImagePhoto(ClientData clientdata, Tcl_Interp *interp, int
 
     if (has_bbox) {
         int srcstride = wdata * 4;
-        destx = x1;
-        desty = hdata - y2;
-        destwidth = x2 - x1;
-        destheight = y2 - y1;
+        destx = (int)x1;
```

Comment:
Shouldn't BBOX_FORMAT just be 4 integers and just use integers all along?  (not sure if tk would have the bad idea to always cast things to float)

Reply:
The call sequence is ``backend.blit(image, bbox)`` -> ``tkagg.py:blit(image, bbox)`` -> ``PyAggImagePhoto(as_string(image, bbox))`` so we cannot change the root ``backend.blit`` call. AFAICT the choice is:
- convert early in ``tkagg.py:blit``
- convert later in ``PyAggImagePhoto``

I usually prefer converting as late as possible to postpone rounding and allow future refactoring with floats to be as painless as possible, but will flow with whatever is matplotlib best practices

Reply:
I think we can leave it as it is for now, we were using floats to start with and we can change that later.